src/Catty/PlayerEngine/Sensors/Object/TouchesEdgeSensor.swift

```diff
@@ -0,0 +1,74 @@
+/**
+ *  Copyright (C) 2010-2021 The Catrobat Team
+ *  (http://developer.catrobat.org/credits)
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  An additional term exception under section 7 of the GNU Affero
+ *  General Public License, version 3, is available at
+ *  (http://developer.catrobat.org/license_additional_term)
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see http://www.gnu.org/licenses/.
+ */
+
+ @objc class TouchesEdgeSensor: NSObject, TouchSensor {
+
+    @objc static let tag = "COLLIDES_WITH_EDGE"
+    static let name = kUIFESensorTouchesEdge
+    static let defaultRawValue = 0.0
+    static let requiredResource = ResourceType.touchHandler
+    static let position = 30
+
+    let getTouchManager: () -> TouchManagerProtocol?
+
+    init(touchManagerGetter: @escaping () -> TouchManagerProtocol?) {
+        self.getTouchManager = touchManagerGetter
+    }
+
+    func tag() -> String {
+        type(of: self).tag
+    }
+
+    func rawValue(for spriteObject: SpriteObject) -> Double {
+        guard let _ = spriteObject.spriteNode else { return type(of: self).defaultRawValue }
+
+        spriteObject.spriteNode.setPhysicsBodyForTouchesEdgeSensor()
+
+        guard let name = spriteObject.spriteNode.name else { return 0 }
+
+        guard let stage = spriteObject.spriteNode.scene as? Stage else { return 0 }
+        if let unwrapped_allContactedBodies = stage.physicsBody?.allContactedBodies() {
+            if stage.physicsBody?.allContactedBodies().count ?? 0 > 0 {
+                if checkForContact(contactedBodies: unwrapped_allContactedBodies, parameter: name) {
+                    return 1
+                }
+            }
+        }
+        return 0
+    }
+
+    func convertToStandardized(rawValue: Double, for spriteObject: SpriteObject) -> Double {
+        rawValue
+    }
+
+    func checkForContact(contactedBodies: [SKPhysicsBody], parameter: String) -> Bool {
+        for index in 0...(contactedBodies.count - 1) where contactedBodies[index].node?.name == parameter {
+            guard contactedBodies[index].node != nil else { return false }
+            return true
+        }
+        return false
+    }
```

Comment:
With the previous change, we can remove this function entirely.

In general, when iterating over an array in Swift, you should use automatic iterators wherever possible, e.g.
```swift
for contactBody in contactedBodies {
    guard contactBody.node != nil else { return false }
}
```
This avoids off-by-one-errors and outOfBounds runtime exceptions, e.g. when the array contains no elements